mapper/pojo/src/main/java/org/hibernate/search/mapper/pojo/bridge/builtin/impl/DefaultJavaUtilDateValueBridge.java

```diff
@@ -23,7 +23,13 @@ public String toString() {
 
 	@Override
 	public Instant toIndexedValue(Date value, ValueBridgeToIndexedValueContext context) {
-		return value == null ? null : value.toInstant();
+		if ( value == null ) {
+			return null;
+		}
+
+		// java.sql.Date does not support toInstant(). See HSEARCH-3670.
+		// We're using here the same strategy applied on DefaultJavaSqlDateValueBridge to handle them.
+		return value instanceof java.sql.Date ? Instant.ofEpochMilli( value.getTime() ) : value.toInstant();
```

Comment:
Couldn't we always use `Instant.ofEpochMilli( value.getTime() )`? That would spare us one `instanceof`.

Reply:
I would have used only as workaround for `java.sql` legacy types. But actually `java.util.Date` is legacy too. Done